Getting started overhaul

rtapella opened the pull request:
Built preview: 
https://docs.maap-project.org/en/getting-started-pr-branch/getting_started.html

Overhaul of all the Getting Started docs. This is a combo of:
- the old GS docs
- Alexey S's Quickstart Guide
- new content and rewritten content
- Emile's rewrite of the conda environment stuff

If you have time please go through the whole thing and try to follow the instructions. Some areas that need review:
- About and the new overview docs
- Jupyter interface overview section of "setting up your account and workspace"
- MAAP storage options section of "setting up your account and workspace"
-  Using maap.py from a Python notebook needs some content to help get started in "Writing code"
- Some helpful starter templates in "Writing code" (Sujen?)
- working with GitLab and GitHub in "Writing code"
- clean up the conda section and replace with Emile's new page if that makes more sense (delete content and reference the new section)
- Registering an algorithm needs testing. Should be able to: 1. clone a demo algorithm 2. edit it 3. register it.
- Run and monitor a job: update with new jobs UI

grallewellyn commented on docs/source/getting_started/getting_started.ipynb:
```diff
@@ -45,19 +46,151 @@
+    "![Login with EarthData](_static/loggingin2.png)\n",
+    "\n",
+    "\n",
+    "3. Enter your “EarthData Login” account credentials here and click “Log in”. You should see a temporary page that says “Redirecting”, followed by the MAAP showing your Workspaces (which will be empty to start):\n",
```
I always get redirected back to 
![Screenshot 2023-05-02 at 7 57 41 AM](https://user-images.githubusercontent.com/114033465/235704738-91d5e6a8-5f84-4272-8e2a-d1cf5f8b8658.png)
Where I need to do tools -> development

grallewellyn replied:
But Rob said: weird. it brings me to the workspace / eclipse-che dashboard

Curious to see what other people see

grallewellyn commented on docs/source/getting_started/getting_started.ipynb:
```diff
@@ -45,19 +46,151 @@
+    " - The Jupyter root directory is mounted to `/projects`.  Stuff in here persists across sessions.\n",
+    "     - Use this for code-related items, smaller data storage\n",
+    "     - Git is more likely to behave predictably here compared to other storage\n",
+    "     - This is also the place to make persistent conda environments (covered in another section), but make sure to not make a conda env inside a git-tracked folder, or if you do add it to the .gitignore. If git is tracking an env, it could cause your workspace to crash. \n",
```
This isn't referred to as the "home" directory here, but in the creating ssh keys section, you refer to the home directory. Should we call this the home directory here to keep the naming more consistent?

rtapella replied:
Yea not sure. I need someone to review because I think there's some historical aspect to this that I'm not familiar with.

grallewellyn commented on docs/source/getting_started/getting_started.ipynb:
```diff
@@ -45,19 +46,151 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Accessing MAAP workspaces over SSH\n",
```
I followed the add SSH steps in OPS and when I try to rename ssh to .ssh, I get an error that .ssh already exists. I press overwrite and I get a 409 error not 404. When I list the contents of .ssh, it only has known hosts, not the authorized_keys file that I just created. I tried this a second time with the same result. If I `ls -al`, I can see both ssh folders listed as like .ssh.15f4974f-888c-4397-9ee9-7cd47f32c346 with my authorized_keys file inside.
If I try deleting the existing .ssh and repeating the steps, I get a 404 and I see my authorized_keys in .ssh
When I click “Display SSH Info”, I see `ssh root@undefined -p undefined`

rtapella replied:
moving this into a different section; it'll need to be re-written and/or bugfixed

grallewellyn commented on docs/source/getting_started/writing_code.ipynb:
```diff
@@ -0,0 +1,308 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With the **File Browser** tab on the JupyterLab sidebar selected, a folder named \"maap-documentation\" should now appear under the \"projects\" directory. Folders for the various sections of the guide can be found in the \"docs/source/\" directory. "
```
Do we want to say home directory instead of "projects" directory to be more consistent with the naming?

grallewellyn replied:
The creating ssh keys section, it is referred to as the home directory. Is that incorrect? 
Also, this section says home directory: 
"NOTE: Custom environments are deleted, and modifications to the base conda environment (e.g., installing mamba) are reset, when the workspace is Stopped. To preserve a custom environment after the workspace shutdowns, create the environment that is saved to the home directory instead of the default system directory for environments. Make sure to not make a conda env inside a git-tracked folder, or if you do add it to the `.gitignore`. If git is tracking an env, it could cause your workspace to crash. This can be done via the following commands, to create the environment at the path `./myenvs/testenv` with the flag `-p` and to activate it using the full path:\n",

rtapella replied:
by "home" I usually think of "~"

wildintellect replied:
sure but on a normal *nix system that ~ would actually go to `/home/username` MAAP is very non-standard in this way so it's somewhat confusing.

rtapella replied:
How about: "Your Jupyter home directory (~) is mounted to /projects. Files in here persist across sessions."

grallewellyn commented on docs/source/getting_started/running_at_scale.ipynb:
```diff
@@ -0,0 +1,148 @@
+    "\n",
+    "### Clone a test algorithm\n",
+    "\n",
+    "**[SHOULD THIS BE IN THE WRITING CODE SECTION?]**\n",
```
I like this here to keep all algorithm related stuff in the same section. I could see a user just coming to read the docs for how to run an algorithm and I know this kinda falls under showing how to use GitHub in the writing code section, but we don't want to be repeating how to do this.
Maybe a solution could be name the header to something like "See GitHub branches and changes" so the user can find it better if they are looking for this information?

rtapella replied:
I feel like maybe "writing code" should have clone-a-repo in it. In terms of a Getting Started we should have a clear path through:
- get your account
- make a workspace
- clone a repo so you have some example code
- change the code ("Writing code") and push it back to github (fork?)
- register "your" new algorithm
- run the algorithm on DPS and watch it finish

I think I should also make a high-level overview of what the Getting Started tutorial will include.. maybe relate it to the overview diagram.

rtapella replied:
will consider for future revision

grallewellyn commented on docs/source/getting_started/writing_code.ipynb:
````diff
@@ -0,0 +1,308 @@
+    "NOTE: Custom environments are deleted, and modifications to the base conda environment (e.g., installing mamba) are reset, when the workspace is Stopped. To preserve a custom environment after the workspace shutdowns, create the environment that is saved to the home directory instead of the default system directory for environments. Make sure to not make a conda env inside a git-tracked fold ...
+    "```\n",
+    "mamba env create -f env.yml -p ./myenvs/testenv\n",
+    "conda  activate ./myenvs/testenv\n",
````
Should we change this line to `conda env create -f newenv2.yml -p ./myenvs/testenv` to show the user it can also be run with conda (we told them above they can just replace conda with mamba). Also, we should continue calling the file newenv2.yml to make it easier to copy and paste

Overall, I am a little confused about the NOTE:. I think we should split this up into 2 different paragraphs. One related to custom environment persistence, and another related to adding it to the .gitignore because this current way makes it seem like those 2 commands will add the conda env to .gitignore
I suggest:
"NOTE: Custom environments are deleted, and modifications to the base conda environment (e.g., installing mamba) are reset, when the workspace is stopped. To preserve a custom environment after the workspace shutdowns, create the environment that is saved to the home directory instead of the default system directory for environments.
```
conda env create -f newenv2.yml -p ./myenvs/testenv
conda  activate ./myenvs/testenv
```
Make sure that the file path in this command for the conda env is not inside a git-tracked folder, or if you do add it to the `.gitignore`. If git is tracking an env, it could cause your workspace to crash. `.gitignore` goes in the root directory of your git repository, and you can just add the file name to this list"

Note that I added information about how to add to a .gitignore because the average user might not be familiar with .gitignore. Even I am not sure what they are adding to the .gitignore. When I ran those commands, no file was created in my myenvs folder. What should they be adding to the .gitignore?

UPDATE: I talked to Rob about this and he thinks the .yml is what we are suggesting not to add to to git, and we should find a best practice for this

grallewellyn replied:
From Anil regarding this: If you are referring to our workspaces? Yes. We have mamba installed in all of them to speed up package installation.

grallewellyn commented on docs/source/getting_started/running_at_scale.ipynb:
```diff
@@ -0,0 +1,148 @@
+    "\n",
+    "### Register the algorithm\n",
+    "\n",
+    "1. Make sure code is ready and saved -> right click file -> “Register as MAS Algorithm”\n",
```
I can't do this right now in DIT by right clicking algorithm_config.yaml, but this might not have been completed yet

rtapella commented on docs/source/getting_started/getting_started.ipynb:
```diff
@@ -45,19 +46,151 @@
+    "![MAAP Tour Notification](_static/maap_tour_notification.png)\n",
+    "\n",
+    "\n",
+    "In addition to [typical JupyterLab menu bar and sidebar](https://jupyterlab.readthedocs.io/en/stable/user/interface.html) configuration:\n",
```
I thought MAAP Login was also being removed from the menubar

grallewellyn replied:
It is! I meant like it is confusing to the user that we are talking about MAAP Login in that list and it isn't in the ADE anymore, I was just giving another reason for why we should remove MAAP Login from the screenshot

rtapella commented on docs/source/getting_started/getting_started.ipynb:
```diff
@@ -45,19 +46,151 @@
+    " - Uses local (not network-based) file system; generally faster and more reliable for “normal” file operations, but expensive\n",
+    "\n",
+    "#### My Private Bucket\n",
+    "`my-private-bucket` is an S3 bucket with persistent storage, but accessible only to you and others in the workspace\n",
```
yes. need to share workspaces. which is only in the FAQ right now

grallewellyn replied:
Should we add that you need to share the workspace?

wildintellect replied:
Yes, the private bucket is assigned by the create/owner of a workspace. If that workspace is shared the private bucket is the owners.

wildintellect commented:
@rtapella I found it easier to read through the actual rendered site, so the comments are not inline with the notebooks, but they are in order and tag the relevant page/section.

https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/maap_overview.html
confusing text, use of DPS before explaining DPS
RESTful API - no regular user needs to know that, that goes in the Reference section

https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/getting_started.html
Callout with note section
Once your MAAP account is approved, you will receive an email notification using the address of your Earthdata Login account to let you know that your access is enabled.

Workspaces are effectively a Jupyter <- JupyterLab 

the info in Stacks is not useful to the user, they want a link to the environments.yml files on github that explain what's in each env


https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/getting_started.html#Jupyter-Interface-overview
Is this not in Jupyter standard docs?


https://docs.maap-project.org/en/getting-started-pr-branch/_images/storageoptions.png
(Custom) Files in the workspace -> is incorrect, the files do exist across workspaces for the workspace owner.


https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/getting_started.html#Sharing-files:-my-public-bucket-and-shared-buckets
doesn't explain the difference between public-bucket (read-write) and shared (read-only)


https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/writing_code.html#Writing-code-with-MAAP
which may be GitHub or MAAP GitLab. <- those are simply the top 2 suggested locations for storing git repos. names like git should be bolded or highlighted

GUI widget is mentioned but terminal is not?
nothing in here about tokens for pushing to gitlab or github

Algorithms run at scale in the MAAP must first be pushed to the MAAP GitLab in order to facilitate registration. <- this might not be true anymore, I think all public repos work in the 3.0 version ask Sujen

https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/writing_code.html#Using-maap.py-to-access-MAAP-functionality-from-Python-notebooks
Doesn't explain what maap.py actually is, and when you might want to use it.


https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/writing_code.html#Helpful-Templates-while-developing-Algorithms-in-MAAP
the current repo does not contain any examples of use to regular users
https://github.com/MAAP-Project/maap-repo-template feels more like an internal MAAP template for devs not users
A separate repo will need to be made for Algorithm template in order to allow use of github Template button. (https://github.com/MAAP-Project/dps-unit-test might be the better template to mention)


https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/running_at_scale.html#Clone-a-test-algorithm
Makes it seem like you have to clone the repo
gets confusing when you get down to https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/running_at_scale.html#Register-the-algorithm

https://docs.maap-project.org/en/getting-started-pr-branch/getting_started/running_at_scale.html#Pop-up
Code example isn't in a code block, impossible to read or copy

rtapella commented:
Thank you Alex. I have implemented most of those changes. I added some new Git Issues in this repo for things I'd like to punt on for now so we can merge this PR.

wildintellect reviewed (approved):
Good first pass for V3. Items not addressed yet are in new tickets #194 #195 #196
